linkerd/distribute/src/lib.rs

```diff
@@ -0,0 +1,258 @@
+//! A load-agnostic traffic distribution stack module.
+
+#![deny(rust_2018_idioms, clippy::disallowed_methods, clippy::disallowed_types)]
+#![forbid(unsafe_code)]
+
+use indexmap::IndexMap;
+use linkerd_stack::{NewService, Param, Service};
+use rand::{
+    distributions::{Distribution as _, WeightedIndex},
+    rngs::SmallRng,
+    SeedableRng,
+};
+use std::{
+    collections::HashSet,
+    hash::Hash,
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+#[derive(Clone, Debug)]
+pub struct NewDistribute<K, S> {
+    backends: Arc<IndexMap<K, S>>,
+}
+
+#[derive(Debug)]
+pub struct Distribute<K, S> {
+    backends: Arc<IndexMap<K, S>>,
```

Comment:
i don't _think_ i actually see anywhere where the ordered nature of the `IndexMap` is used, is that correct?

(but see my comment later where i discuss a way where we could use `IndexMap` to our advantage if handling services that don't share readiness across clones is desirable...)

Reply:
yeah, this was a remnant of storing indices instead of service clones...